apps/api/src/domain/models/digital-text/commands/create-digital-text.command-handler.ts

```diff
@@ -0,0 +1,146 @@
+import { AggregateId } from '../../../types/AggregateId';
+import { InMemorySnapshot, ResourceType } from '../../../types/ResourceType';
+import buildInMemorySnapshot from '../../../utilities/buildInMemorySnapshot';
+import { BaseCommandHandler } from '../../shared/command-handlers/base-command-handler';
+import UuidNotGeneratedInternallyError from '../../shared/common-command-errors/UuidNotGeneratedInternallyError';
+import { BaseEvent } from '../../shared/events/base-event.entity';
+import { DigitalText } from '../digital-text.entity';
+import { CreateDigitalText } from './create-digital-text.command';
+import { DigitalTextCreated } from './digital-text-created.event';
+
+@CommandHandler(CreateDigitalText)
+export class CreateDigitalTextCommandHandler extends BaseCommandHandler<DigitalText> {
+    protected repositoryForCommandsTargetAggregate: IRepositoryForAggregate<DigitalText>;
+
+    constructor(
+        @Inject(REPOSITORY_PROVIDER_TOKEN)
+        protected readonly repositoryProvider: IRepositoryProvider,
+        @Inject('ID_MANAGER') protected readonly idManager: IIdManager
+    ) {
+        super(repositoryProvider, idManager);
+
+        this.repositoryForCommandsTargetAggregate =
+            this.repositoryProvider.forResource<DigitalText>(ResourceType.digitalText);
+    }
+
+    protected async createOrFetchWriteContext({
+        aggregateCompositeIdentifier: { id },
+        title,
+        languageCodeForTitle,
+    }: CreateDigitalText): Promise<ResultOrError<DigitalText>> {
+        const digitalTextDto: DTO<DigitalText> = {
+            id,
+            title: buildMultilingualTextWithSingleItem(title, languageCodeForTitle),
+            published: false,
+            type: ResourceType.digitalText,
+            eventHistory: [],
+        };
+
+        return getInstanceFactoryForResource<DigitalText>(ResourceType.digitalText)(digitalTextDto);
+    }
+
+    actOnInstance(digitalText: DigitalText): ResultOrError<DigitalText> {
+        return digitalText;
+    }
+
+    protected async fetchRequiredExternalState(): Promise<InMemorySnapshot> {
+        const searchResult = await this.repositoryProvider
+            .forResource<DigitalText>(ResourceType.digitalText)
+            .fetchMany();
+
+        const allDigitalTexts = searchResult.filter((digitalText): digitalText is DigitalText => {
+            if (isInternalError(digitalText)) {
+                throw digitalText;
+            }
+
+            return true;
+        });
+
+        // Deal with deprecation? Use DeluxeInMemoryStore?
+        return buildInMemorySnapshot({
+            resources: { digitalText: allDigitalTexts },
+        });
+    }
+
+    validateExternalState(state: InMemorySnapshot, digitalText: DigitalText): ValidationResult {
+        return digitalText.validateExternalState(state);
+    }
+
+    /**
+     * Where should we do this? It seems natural to do this as part of the
```

Comment:
I think you are using the wrong command handler for "inspiration". My advice is always to look at the most recent similar command. But definitely look around and be careful not to fall victim to drift randomization. There are different ways of doing this now. Another thing that might help is to start following Blake's command PRs more closely.